sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1555,9 +1554,25 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     try {
       valueType match {
         case "DATE" =>
-          Literal(Date.valueOf(value))
+          val castedValue = Cast(
+            Literal(value),
+            DateType,
+            Some(SQLConf.get.sessionLocalTimeZone)).eval()
```

Comment:
Please use `Option(SQLConf.get.sessionLocalTimeZone)`.

Reply:
I re-implemented it without `Option`